src/app/[locale]/specs/sync/page.mdx

```diff
@@ -0,0 +1,171 @@
+
+If `since` is not included, the commit should include the full repo tree, or set the `tooBig` flag.
+
+If the `tooBig` flag is set, then the amount of updated data was too much to be serialized in a single stream event message. Downstream services which want to maintain complete synchronized copies for the repo need to fetch the diff separately, as discussed below.
+
+### Firehose Validation Best Practices
+
+A service which does full validation of upstream events has a number of properties to track and check. For example, Relay instances should fully validation content from PDS instances before re-broadcasting.
+
+Here is a summary of validation rules and behaviors:
+
+- services should independently resolve identity data for each DID. They should ignore `#commit` events for accounts which do not have a functioning atproto identity (eg, lacking a signing key, or lacking a PDS service entry, or for which the DID is tombstoned)
+- services which subscribe directly to PDS instances should keep track of which PDS is authoritative for each DID. They should remember the host each subscription (WebSocket) is connected to, and reject `#commit` events for accounts if they come from a stream which does not correspond to the current account for that DID
+- services should track account hosting status for each DID, and ignore `#commit` events for events which are not `active`
+- services should verify commit signatures for each `#commit` event, using the current identity data. If the signature initially fails to verify, the service should refresh the identity metadata in case it had recently changed. Events with conclusively invalid signatures should be rejected.
+- services should reject any event messages which exceed reasonable limits. A reasonable upper bound for producers is 5 MBytes (for any event stream message type). Commits with too much data must use the `tooBig` mechanism, though such commits should generally be avoided in the first place by breaking them up in to multiple smaller commits.
+- services should verify that repository data structures are valid against the specification. Missing fields, incorrect MST structure, or other protocol-layer violations should result in events being rejected.
+- services may apply rate-limits to identity, account, and commit events, and suspend accounts or upstream services which violate those limits. Rate limits might also be applied to recovery modes such as invalid signatures resulting in an identity refresh, `tooBig` events, missing or out-of-order commits, etc.
+- services should ignore commit events with a `rev` lower or equal to the most recent processed `rev` for that DID, and should reject commit events with a `rev` corresponding to a future timestamp (beyond a clock drift window of a few minutes)
+- services should check the `since` value in commit events, and if it does not match the previous seen `rev` for that DID, mark the repo as out-of-sync (similar to a `tooBig` commit event)
```

Comment:
Is there a nuance here, i.e. could `since` be `<=` previous rev?  IIRC this would at least be in line with the repo diffs.  Though at the same time, we don't necessarily need to permit it here.

Reply:
this was discussed in more detail in "Reliable Synchronization". I updated the text to say "consistent with" instead of "match", and referenced that section for details.